contents/testing/integration-testing.md

```diff
@@ -0,0 +1,143 @@
+| **Advantages** | Early discovery of high level architecture / design defects | Easier to create test cases bottom up | Beneficial for big project to distribute tasks on testing
+| | Main control points of the system are tested early | Critical modules on functionalities ae tested first | Allow top-down and bottom-up approach to run side by side | 
+| **Disadvantages** | Significant low level moduls are tested late in the cycle | There is no working system until the last module is build | It is difficult to test for highly interconnected modules |
+| | A [<span title="A program that simulates the behaviours of software components">stub</span>](https://stackoverflow.com/questions/463278/what-is-a-stub) is not perfect to simulate data flow | A [<span title="Module with dummy code to temporalities replace a module">driver</span>](http://www.professionalqa.com/test-driver) test is even harder to write than stub | Higher cost from using both driver and stub. You can understand the difference [here](https://www.quora.com/What-is-the-difference-between-stubs-and-drivers-in-software-testing)
+
+:information_source: A more detailed guide on using specific methods for incremental testing together with examples can be found [here](https://www.softwaretestinghelp.com/incremental-testing/).
+
+### Designing test scenarios, cases and scripts
+
+Before the actual coding is done, a basic test strategy deciding the test cases and test data used should be crafted. This usually involves setting a <span style="color:red">test case ID</span>, <span style="color:red">objective</span>, <span style="color:red">description</span> and <span style="color:red">expected result</span>. Using the example shown above, below shows a sample integration test used for the login and ordering modules:
+
+~~~
+Test case ID: 1
+Objective: Check the link between login and ordering modules
+Description: Enter login credentials and click on login button
+Expected result: To be directed to order food page based on the login user
+~~~
+
+<box type="warning">As integration test cases are expensive operations compared to unit testing, it should focus mainly on the integration of modules together and not on specific actions within the same module.</box>
+
+:information_source: More details about the ways to structure incremental testing can be found on [this article](https://www.softwaretestinggenius.com/various-approaches-in-integration-testing/)
+
+### Executing test scripts and report defects
+
+Depending on the approach you have chosen for your integration plan and the test cases, the way you execute your code for testing will differ.
+
+A big bang approach usually involves all the modules to be developed before you can start with the integration testing.
+
+As for incremental approach, it is usually conducted simultaneously with the modules development. Hence, stubs and drivers are used to mimic the modules for writing tests.
```

Comment:
modules`'` development

Reply:
Actually might be better to write `with the development of modules.` or `with the module's development.`  😅 

If you go with the latter, you'll need to update the next sentence as well.


May also be good to add `mimic the modules for writing tests`**`since...`** and remove `Hence`?